### PR TITLE
added BatMap.find_default

### PR DESCRIPTION
**Pull request by UnixJunkie**
implementing https://github.com/ocaml-batteries-team/batteries-included/issues/723

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -3,6 +3,9 @@ Changelog
 
 ## NEXT_RELEASE (minor release)
 
+- BatMap: added find_default
```
Comment (gasche) [inline ChangeLog]: Having the GPR number would be nice.

File: src/batMap.ml [hunk 2]
```diff
@@ -944,6 +950,9 @@ let find x m = Concrete.find x Pervasives.compare m
   empty |> add 2 'y' |> add 1 'x' |> find 2 = 'y'
 *)
 
+let find_default def x m =
+  Concrete.find_default def x Pervasives.compare m
```
Comment (gasche) [inline src/batMap.ml]: Having two unit tests (the default and the non-default cases) would be nice.

**Comment by UnixJunkie** [comment]
I just pushed for the pull request number in the ChangeLog.

**Comment by UnixJunkie** [comment]
two more unit tests added also

**Comment by UnixJunkie** [comment]
this is ready to merge I think

**Pull request merged**